### DISCUSSION TITLE
Relation RULE for users in the same group

### DISCUSSION THREAD

**Discussion by srowe0091** [pocketbase/pocketbase#5709] [Q&A] [answered]
Hello, I added screenshots of what my table looks like to see if it will help with the rule i need to get the behavior I want

### Collections table
![collections](https://github.com/user-attachments/assets/7b32e489-9d40-4399-96ec-693fcafd1eb5)

Each collection has an owner and a name

---

### Cards table
![cards](https://github.com/user-attachments/assets/6bdc966e-904c-42c6-8754-30cc097577f2)

Each card is in a collection

---
### Groups table
![Groups](https://github.com/user-attachments/assets/2e0d3c0f-3402-4bd0-9074-e69836845fc7)

Each group just has a name and who created the group

---
### Groups Users table

![GroupsUsers](https://github.com/user-attachments/assets/f7cd5271-154d-4eff-9583-d5ecb9a6039f)

This is where the relationships happen between users

---

What i need is a rule for COLLECTIONS and CARDS to be able to be only VIEWED by users that are in the same group.

So using the tables above, user `test` should be able to VIEW the collection (to get the name) and the cards in that collection for the `test 2` user and vice versa, `test 2` should be able to see `test` because they are both in the same group.

Since `test 3` is not in any group with `test` or `test 2` then no one is allowed to be able to view `test 3` collection or cards.

I would love any advise as I'm not too experienced with databases and what the best way it is to create relationships with what I am trying to achieve. I'm still early in this so if i need to change my data structure, that wont be an issue

**Comment by pnmcosta** [comment]
I think your answer may be here https://github.com/pocketbase/pocketbase/discussions/3805#discussioncomment-7634791

**Comment by ganigeorgiev** [comment, marked as answer]
_Please avoid posting screenshots as it is difficult to read and reproduce locally. You can export your collections configuration as json from the Admin UI > Settings > Export collections._

Unless I misunderstood your db structure the bellow "collections" View API rule should suffice:

```js
@request.auth.id != "" &&
// at least one of the current authenticated user groups matches with at least one of the linked collections' owner
@request.auth.groupsUsers_via_userId.groupId ?= ownerId.groupsUsers_via_userId.groupId
```

For the "cards" is similar but you have to go through "collections":

```js
@request.auth.id != "" &&
@request.auth.groupsUsers_via_userId.groupId ?= collectionsId.ownerId.groupsUsers_via_userId.groupId
```

(_in both cases the API rules rely on the [back-relations](https://pocketbase.io/docs/working-with-relations/#back-relations) filter syntax_).

**Reply by srowe0091** [reply]
Thank you! those rules worked. sorry for the screenshots. I'll be sure to post the collection configurations next time I need help.

I knew it had something to do with back-relations. But I can't wrap my head around what the rules look like for that to make things work like this.

**Reply by dschissler** [reply]
@srowe0091 It's only purely cosmetic but consider having your collection relation fields without the `Id` suffix.  I was always doing that before, but with Pocketbase I like it without because of how fluid the relation filter dot notation is.
